libs/core/langchain_core/stores.py

```diff
@@ -123,3 +129,292 @@ async def ayield_keys(
+            )
+
+        return Path(full_path)
+
+    def _mkdir_for_store(self, dir: Path) -> None:
+        """Makes a store directory path (including parents) with specified permissions
+
+        This is needed because `Path.mkdir()` is restricted by the current `umask`,
+        whereas the explicit `os.chmod()` used here is not.
+
+        Args:
+            dir: (Path) The store directory to make
+
+        Returns:
+            None
+        """
+        if not dir.exists():
+            self._mkdir_for_store(dir.parent)
+            dir.mkdir(exist_ok=True)
+            if self.chmod_dir is not None:
+                os.chmod(dir, self.chmod_dir)
+
+    def mget(self, keys: Sequence[str]) -> List[Optional[bytes]]:
+        """Get the values associated with the given keys.
+
+        Args:
+            keys: A sequence of keys.
+
+        Returns:
+            A sequence of optional values associated with the keys.
+            If a key is not found, the corresponding value will be None.
+        """
+        values: List[Optional[bytes]] = []
+        for key in keys:
+            full_path = self._get_full_path(key)
+            if full_path.exists():
+                value = full_path.read_bytes()
+                values.append(value)
+            else:
+                values.append(None)
+        return values
+
+    def mset(self, key_value_pairs: Sequence[Tuple[str, bytes]]) -> None:
+        """Set the values for the given keys.
+
+        Args:
+            key_value_pairs: A sequence of key-value pairs.
+
+        Returns:
+            None
+        """
+        for key, value in key_value_pairs:
+            full_path = self._get_full_path(key)
+            self._mkdir_for_store(full_path.parent)
+            full_path.write_bytes(value)
+            if self.chmod_file is not None:
+                os.chmod(full_path, self.chmod_file)
+
+    def mdelete(self, keys: Sequence[str]) -> None:
+        """Delete the given keys and their associated values.
+
+        Args:
+            keys (Sequence[str]): A sequence of keys to delete.
+
+        Returns:
+            None
+        """
+        for key in keys:
+            full_path = self._get_full_path(key)
+            if full_path.exists():
+                full_path.unlink()
+
+    def yield_keys(self, prefix: Optional[str] = None) -> Iterator[str]:
+        """Get an iterator over keys that match the given prefix.
+
+        Args:
+            prefix (Optional[str]): The prefix to match.
+
+        Returns:
+            Iterator[str]: An iterator over keys that match the given prefix.
+        """
+        prefix_path = self._get_full_path(prefix) if prefix else self.root_path
+        for file in prefix_path.rglob("*"):
+            if file.is_file():
+                relative_path = file.relative_to(self.root_path)
+                yield str(relative_path)
+
+
+class InMemoryBaseStore(BaseStore[str, V], Generic[V]):
```

Comment:
this is fine in core too. But if we're doing in memory implementations of everything, could also put all these in a `langchain-inmemory` implementation package

Reply:
IMO core might make more sense since the in memory abstraction is useful for testing of core itself